ccpp/data/GFS_typedefs.F90

```diff
@@ -4977,6 +5000,13 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     Model%restart          = restart
     Model%lsm_cold_start   = .not. restart
     Model%hydrostatic      = hydrostatic
+
+    if(Model%hydrostatic .and. Model%lightning_threat) then
```

Comment:
I assume with this setting, we do not need to pass hydrostatic to ccpp phys diagnostic subroutine?

Reply:
My understanding is that the three lightning threat indexes will not be allocated, please see line 7156, and the if statements of filling the index arrays to zero in ccpp physics will not be called. In other words, the lightning threat indexes will not be output to history files.

Reply:
My worry is that `hydrostatic` will change value, since this diagnostic print says it can:

```
      print *, 'variables modified at each time step'
      ...
      print *, ' hydrostatic       : ', Model%hydrostatic
```

Reply:
I see, Thanks. It's interesting that this "hydrostatic" variable could change during integration. @ChunxiZhang-NOAA @grantfirl Can you confirm that the hydrostatic variable could change during model integration?

Reply:
I don't think "hydrostatic" variable can be changed during integration.

Reply:
I don't think it makes sense or is possible to change between hydrostatic and non-hydrostatic during integration. I also see no evidence of hydrostatic being set anywhere outside of initialization.

Reply:
For my own sanity, I'm moving "hydrostatic" up to the "basic control parameters" in the control_print.

Reply:
@SamuelTrahanNOAA Are you planning to make any changes?

Reply:
The hydrostatic check is removed from the ccpp/physics code, and the hydrostatic print is in the right section.